pkg/platform/kube/test/suite.go

```diff
@@ -118,6 +123,59 @@ func (suite *KubeTestSuite) TearDownTest() {
 	suite.Require().NoError(err)
 }
 
+func (suite *KubeTestSuite) deployAPIGateway(createAPIGatewayOptions *platform.CreateAPIGatewayOptions,
+	onAfterIngressCreated OnAfterIngressCreated,
+	expectError bool) {
+
+	// deploy the api gateway
+	err := suite.Platform.CreateAPIGateway(createAPIGatewayOptions)
+
+	if !expectError {
+		suite.Require().NoError(err)
+	} else {
+		suite.Require().Error(err)
+	}
+
+	// delete the api gateway when done
+	defer func() {
+
+		if err != nil {
+			suite.Logger.Debug("Deleting deployed api gateway")
+			err := suite.Platform.DeleteAPIGateway(&platform.DeleteAPIGatewayOptions{ // nolint: errcheck
+				Meta: createAPIGatewayOptions.APIGatewayConfig.Meta,
+			})
+			suite.Require().NoError(err)
+		}
+	}()
+
+	// give the ingresses some time - after 10 seconds, give up
+	deadline := time.Now().Add(10 * time.Second)
+
+	var ingress *extensionsv1beta1.Ingress
+
+	for {
+
+		// stop after 10 seconds
+		if time.Now().After(deadline) {
+			suite.FailNow("API gateway ingress didn't create in time")
+		}
+
+		ingress, err = suite.KubeClientSet.
+			ExtensionsV1beta1().
+			Ingresses(suite.Namespace).
+			Get(
+				// TODO: consider canary ingress as well
+				kube.IngressNameFromAPIGatewayName(createAPIGatewayOptions.APIGatewayConfig.Meta.Name, false),
+				metav1.GetOptions{})
+		if err == nil {
+			suite.Logger.DebugWith("API gateway ingress found")
+			break
+		}
+	}
+
+	onAfterIngressCreated(ingress)
```

Comment:
This is the pattern used in the rest of the tests
It is done so that I can `defer` a call to the deletion function at the end of the `deployAPIGateway`, If I'll do what you suggested the callee will need to delete the API Gateway